### PR TITLE
feat: integrate `recase` and include `pascalDotCase`

### PR DESCRIPTION
**Pull request by LukeMoody01**
## Status

**READY**

This PR adds support for additional string casing styles, specifically Title.Case (PascalDotCase) #706 , to the Mason package. As ReCase is no longer maintained (My PR has been open for quite a while), I have moved the package over to mason and documented the class. I have updated the corresponding string extensions and render functions to support the new format.

Let me know if there are any issues with this approach! I also feel this could give developers the options to override / create their own LambdaFunctions which would be cool!

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by renancaraujo — APPROVED** [review]

File: packages/mason/lib/src/string_case_extensions.dart [hunk 1]
```diff
@@ -23,6 +23,9 @@ extension StringCaseExtensions on String {
   /// PascalCase
   String get pascalCase => ReCase(this).pascalCase;
 
+  /// Title.Case
```
Comment (felangel) [inline packages/mason/lib/src/string_case_extensions.dart:26]:
```suggestion
  /// Pascal.Dot.Case
```

File: packages/mason/lib/src/render.dart [hunk 2]
```diff
@@ -45,6 +45,9 @@ final _builtInLambdas = <String, LambdaFunction>{
   /// PascalCase
   'pascalCase': (ctx) => ctx.renderString().pascalCase,
 
+  /// Title.Case
```
Comment (felangel) [inline packages/mason/lib/src/render.dart:48]:
```suggestion
  /// Pascal.Dot.Case
```

File: packages/mason/lib/src/recase.dart [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  /// Returns the text in param-case format.
+  String get paramCase => _getSnakeCase(separator: '-');
+
+  /// Returns the text in path/case format.
+  String get pathCase => _getSnakeCase(separator: '/');
+
+  /// Returns the text in PascalCase format.
+  String get pascalCase => _getPascalCase();
+
+  /// Returns the text in PascalDotCase format.
```
Comment (felangel) [inline packages/mason/lib/src/recase.dart:70]:
```suggestion
  /// Returns the text in Pascal.Dot.Case format.
```

**Review by felangel — APPROVED** [review]
LGTM thanks so much for the contribution! 💯

**Pull request merged**